core/frontend/src/Viewport.ts

```diff
@@ -3121,15 +3122,32 @@ export class ScreenViewport extends Viewport {
       assert(undefined === this._webglCanvas); // see getter...
       this._webglCanvas = webglCanvas;
 
-      // this.canvas has zIndex 10. Make webgl canvas' zIndex lower so that canvas decorations draw on top.
       this.addChildDiv(this.vpDiv, webglCanvas, 5);
+
+      /** The following workaround resolves an issue specific to iOS Safari. We really want this webgl canvas' zIndex to be
+       * lower than this.canvas, but if we do that on iOS Safari, Safari may decide to not display the canvas contents once
+       * it is re-added to the parent div after dropping other viewports. It will only display it once resizing the view.
+       * The offending element here is the 2d canvas sitting on top of the webgl canvas. We need to clear its contents
+       * immediately on iOS. Even though the 2d canvas gets cleared in OnScreenTarget.drawOverlayDecorations() in this case,
+       * it looks like iOS needs an immediate clear.
+       */
+      if (System.instance.isMobile)

```

Comment:
@pmconne This should change based on Keith's changes - it ideally should just be iOS.  Who knows if Android has the same problem, but should be evaluated separately.  Think we should PR this as is or wait for Keith's changes?

Reply:
No need to wait - his changes are merged to master.
In a follow-up PR we should remove detectIsMobile() and RenderSystem.isMobile (and I think TileAdmin has an `isMobile` too?)

Reply:
@pmconne Ah - good. I'll change this to use his new method to detect just iOS and verify it still works.  Try to merge this PR in tomorrow.

Reply:
@pmconne I'm looking at core/mobile-manager/src/common/MobileUtils.ts which contains a ProcessDetector class - but there is another file core/Bentley/src/ProcessDetector.ts which contains a similarly named class.  One is public, one is internal.  They seem to have a lot of duplicate code.  Any idea which is preferred to use?
Do you understand how the API differentiates between browser and app frontends? ... will we have to call something like the following?

`const isMobileAppleDevice = ProcessDetector.isIOSBrowser || ProcessDetector.isIpadBrowser || ProcessDetector.isIOSAppFrontend;`

Reply:
I think MobileUtils.ts was supposed to be removed. I also think `ProcessDetector.isIOSFrontend` would be useful. @kabentley @swbsi.

Reply:
yes,` MobileUtils.ts` was cruft that was supposed to be removed in my final merge.

I think you want `ProcessDetector.isIOSBrowser`

Reply:
Will that return true for an iOS app like SYNCHRO Field? The name and description suggest it will not.

Reply:
i will now make a PR to remove MobileUtils.ts and to clarify the documentation (that the frontend of a mobile app *is* a browser).

Reply:
@markschlosseratbentley can i add this to your PR?

Reply:
@kabentley sure thing.

Reply:
@pmconne @markschlosseratbentley see what you think. 

I didn't add methods to detect user-launched mobile browsers, because i didn't think you'd need them. But, that would be `ProcessDetector.isIOSBrowser && !ProcessDetector.isIOSAppFrontend`. Is that OK?

Reply:
BTW, `isIOSBrowser` is supposed to return `true` for ipads, although I confess I don't understand what the second half of `isIpadBrowser` does, i got that from you.

Reply:
Mark's original `detectIsMobile` contained a link to the source from which his code was derived - I suggest we add that to ProcessDetector.
  // Modified from package 'detect-gpu': https://github.com/TimvanScherpenzeel/detect-gpu/blob/master/src/index.ts

Reply:
@kabentley Ok, I tested this out, and isIOSBrowser is not returning true when I run display-test-app on my iPad.  There is a case to be made to differentiate between iOS (phones) and iPadOS (iPads) as Apple does themselves.  I could probably work around this by just calling isIOSBrowser || isIpadBrowser.

Could also add a third property - isMobileAppleBrowser - which returns isIOSBrowser || isIpadBrowser.

Reply:
i think people will expect `isIOSBrowser` to return `true` for iPads. Let me add that (and the attribution). Unfortunately I don't have an iPad set up for testing, so I'll wing it and you can test.

Reply:
@markschlosseratbentley can you try now?

Reply:
i added isIPhoneBrowser, (and changed the case of isIPadBrowser)

Reply:
@kabentley Works on iPad - and the code looks good to me.